include/seqan3/search/configuration/max_error.hpp

```diff
@@ -70,9 +70,34 @@ class max_error_substitution : public pipeable_config_element<max_error_substitu
  * ### Example
  * \include test/snippet/search/configuration_error.cpp
  */
-class max_error_insertion : public pipeable_config_element<max_error_insertion, std::variant<error_count, error_rate>>
+struct max_error_insertion : public pipeable_config_element<max_error_insertion>
 {
-public:
+    //!\brief  The number of insertion errors allowed for the search.
+    std::variant<error_count, error_rate> error{error_count{0}};
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr max_error_insertion() = default; //!< Defaulted.
+    constexpr max_error_insertion(max_error_insertion const &) = default; //!< Defaulted.
+    constexpr max_error_insertion(max_error_insertion &&) = default; //!< Defaulted.
+    constexpr max_error_insertion & operator=(max_error_insertion const &) = default; //!< Defaulted.
+    constexpr max_error_insertion & operator=(max_error_insertion &&) = default; //!< Defaulted.
+    ~max_error_insertion() = default; //!< Defaulted.
+
+    /*!\brief Initialises the insertion error with the given seqan3::search_cfg::error_count.
+     * \param[in] error The error count used for the search.
+     */
+    constexpr explicit max_error_insertion(error_count error) : error{std::move(error)}
+    {}
+
+    /*!\brief Initialises the insertion error with the given seqan3::search_cfg::error_rate.
+     * \param[in] error The error rate used for the search.
```

Comment:
It would be nice to specify the difference of count and rate for the user here or simply express in different words. We often have documentation in the same words as the variable name and it does not add any helpful information in this way. I also guess that there is a fundamental difference in the count being an unsigned integral number and a rate should be a floating point number in range [0,1]?

Reply:
that is documented on the class itself.

Reply:
```suggestion
     * \param[in] error The insertion error rate allowed in the search.
```